### PR TITLE
Add device updates example.y

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: example/src/DeviceUpdates.hs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+  ) \\ (iinstF @'DeviceRef Proxy :: Ords' refs 'DeviceRef)
+    \\ (iinstF @'UpdateRef Proxy :: Ords' refs 'UpdateRef)
+
+postconditions
+  :: forall refs resp. IxForallF Ord refs
+  => Model refs -> Action refs resp -> Response_ refs resp -> Property
+postconditions m cmd resp = (case cmd of
+  ViewUpdates dref -> resp === installedUpdates m M.! dref
+  PullUpdates dref -> pendingUpdates m' M.! dref === S.empty
```
Comment (Danten) [inline example/src/DeviceUpdates.hs]:
```
UpdateReports dref -> resp === installedUpdates m M.! dref
```

File: example/src/DeviceUpdates.hs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  , (5, return . Untyped $ PullUpdates SDeviceRef)
+  , (5, return . Untyped $ ViewUpdates SDeviceRef)
+  ]
+
+shrink1 :: Action refs resp -> [Action refs resp]
+shrink1 _ = []
+
+------------------------------------------------------------------------
+
+semantics
```
Comment (Danten) [inline example/src/DeviceUpdates.hs]: Maybe another name for this to signal that this is not the "real" semantics, `debugSemantics` or something like that.

File: example/src/DieHard.hs [hunk 3]
```diff
@@ -1,6 +1,8 @@
-{-# LANGUAGE DataKinds  #-}
-{-# LANGUAGE GADTs      #-}
-{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeInType            #-}
```
Comment (Danten) [inline example/src/DieHard.hs]: Ehh, These additions are not necessary right?

**Comment by stevana** [comment]
Fixed all comments, thanks!

**Pull request merged**